File: backend/JavaSpringBoot/egypthomes/src/main/java/com/egypthomes/dto/UserDTO.java
package com.egypthomes.dto;

import io.micrometer.common.lang.Nullable;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@NoArgsConstructor
@AllArgsConstructor
@Data
@Builder
public class UserDTO {
        @NotBlank
        Long id;
        @NotBlank
        @Email
        String email;
        @NotBlank
        String firstName;
        @NotBlank
        String lastName;
        @Nullable
        String phoneNumber;
        @Nullable
        String pictureUrl;
}

File: backend/ASP.NET/EgyptHomes/DTOs/UserDTO.cs
using EgyptHomes.Models;

namespace EgyptHomes.DTOs;
public record UserDTO
{
    public required string Id { get; set; }
    public required string Email { get; set; }
    public required string FirstName { get; set; }
    public required string LastName { get; set; }
    public string? PictureUrl { get; set; }
    public string? PhoneNumber { get; set; }
}

public static class UserDTOExtensions
{
    public static UserDTO ToDTO(this User user)
    {
        return new UserDTO
        {
            Id = user.Id,
            Email = user.Email!,
            FirstName = user.FirstName,
            LastName = user.LastName,
            PictureUrl = user.PictureUrl,
            PhoneNumber = user.PhoneNumber
        };
    }
}